src/main/java/eu/arrowhead/application/skeleton/consumer/ConsumerMain.java

```diff
@@ -0,0 +1,397 @@
+		} catch (IOException e) {
+			logger.error(e.getMessage());
+			return null;
+		}
+		List<AuthRule> rules = new ArrayList<>();
+
+		for (JsonNode rule : newRulesJson) {
+			try {
+				AuthRule newRule = Utilities.fromJson(rule.toString(), AuthRule.class);
+				// Setting the defaults if no interfaces are given
+				if (newRule.getInterfaces().size() == 0) {
+					newRule.setInterfaces(new ArrayList<String>(defaultInterfaceNames));
+				}
+				rules.add(newRule);
+			} catch (Exception e) {
+				logger.error("Could not create authorization rule (invalid json format): " + e.getMessage());
+			}
+		}
+		return rules;
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private void updateAuthRules(final List<AuthRule> newRules) {
+		deleteRules(newRules);
+		addRules(newRules);
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private void deleteRules(final List<AuthRule> rules) {
+		List<Integer> ruleIdsToDelete = new ArrayList<Integer>();
+		try {
+			ruleIdsToDelete = getRuleIdsToDelete(getAuthorizationRules(), getSystemIdsToDelete(rules));
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		}
+
+		Map<String, String> authorizationUri = null;
+		try {
+			authorizationUri = getAuthorizationUri();
+		} catch (Exception e) {
+			logger.error(e.getMessage());
+		}
+
+		for (int id : ruleIdsToDelete) {
+			logger.info("Removing authorization rule with id: " + id);
+			String response = arrowheadService.consumeServiceHTTP(String.class, HttpMethod.DELETE,
+					Utilities.createURI(authorizationUri.get("scheme"), authorizationUri.get("host"),
+							Integer.parseInt(authorizationUri.get("port")), authorizationUri.get("path")
+									+ ConsumerConstants.QUERY_AUTH_INTRA_CLOUD + "/" + Integer.toString(id)),
+					null, null);
+			logger.info("Http DELETE response: " + response);
+		}
+	}
+
+	// -------------------------------------------------------------------------------------------------
+	private void addRules(final List<AuthRule> rules) {
+
+		List<AuthorizationIntraCloudRequestDTO> rulesToAdd = createDTOListFromAuthRules(rules);
+		for (AuthorizationIntraCloudRequestDTO ruleToAdd : rulesToAdd) {
+			addSingleRule(ruleToAdd);
+		}
+	}
+
+	// =================================================================================================
+	// assistant methods
+
+	// -------------------------------------------------------------------------------------------------
+	private List<AuthorizationIntraCloudRequestDTO> createDTOListFromAuthRules(final List<AuthRule> rules) {
+
+		List<AuthorizationIntraCloudRequestDTO> dtoList = new ArrayList<>();
+		Long consumerId = null;
+		List<Long> providerId;
+		List<Long> serviceDefinitionId;
+		List<Long> interfaceIds;
+
+		for (int i = 0; i < rules.size(); i++) {
+
+			providerId = new ArrayList<>();
+			serviceDefinitionId = new ArrayList<>();
```

Comment:
these lists are always contains only one element. You should set the initial capacity to 1 in the constructor.